arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,40 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int TRIGGER_PIN           = 6; 
+const int ECHO_PIN              = 7; 
+const unsigned int MAX_DISTANCE = 200;
+const int FRONT_PIN = 0;
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+GP2Y0A21 frontIR(arduinoRuntime, FRONT_PIN); 
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+    car.setSpeed(30);
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+  int usDistance = front.getDistance();
+  int irDistance = frontIR.getDistance();
+  if((usDistance < 170) && (usDistance > 30))
+  {
+    car.setAngle(90);
+    car.setSpeed(15);
+    delay(3000);
+    car.setAngle(0);
+    car.setSpeed(30);
+    usDistance = front.getDistance();
```

Comment:
Now that i'm looking at this carefully, this isn't needed either :sweat_smile:

Reply:
We tested it just now and it works the same without it. We had some problems before and thought that maybe the distances didn't get updated.

Reply:
I mean, the new value is never used as it goes out of scope and a new variable is crated in the next `loop` iteration, so I would be extremely surprised if this had any effect on the behavior you're observing :sweat_smile: